warehouse/internal/repo/schema.go

```diff
@@ -134,6 +154,27 @@ func (sh *WHSchema) Insert(ctx context.Context, whSchema *model.WHSchema) error
 
 		// If table-level schema is enabled, insert/update for each table
 		if sh.config.enableTableLevelSchema.Load() {
+			if len(whSchema.Schema) > 0 {
+				// Delete orphaned table-level schemas for the current source_id + destination_id + namespace
+				// This ensures consistency between raw schema and table-level schemas
+				_, err = tx.ExecContext(ctx, `
+				DELETE FROM `+whSchemaTableName+`
+				WHERE destination_id = $1
+				  AND namespace = $2
+				  AND table_name != ''
+				  AND table_name NOT IN (
+					SELECT unnest($3::text[])
+				  );
+			`,
+					whSchema.DestinationID,
+					whSchema.Namespace,
+					pq.Array(lo.Keys(whSchema.Schema)),
+				)
```

Comment:
@achettyiitr 
With SchemaTTL, this query would run less frequently. Query should be fast, but `table_name NOT IN` means it will scan all table_name rows from the table.
If I am right, this cleanup we are doing to match table-level schema and wh-schema. No issue with stale tables schema? As in we can think of optimizing cleaning stale tables schema if needed.

WDYT? If yes, I will add a task in backlog and add reference.